File: program/program/implement-selection-sort/ImplementSelectionSort.rb
def selection_sort(arr)
  n = arr.length

  for i in 0..n-1
    min_index = i

    for j in (i+1)..n-1
      min_index = j if arr[j] < arr[min_index]
    end

    arr[i], arr[min_index] = arr[min_index], arr[i] if min_index != i
  end
end


puts "Enter numbers separated by spaces:"
user_input = gets.chomp
numbers = user_input.split.map(&:to_i)

puts "Original array: #{numbers}"

selection_sort(numbers)

puts "Sorted array: #{numbers}"

File: program/program/implement-selection-sort/implement_selection_sort.rs
fn selection_sort(arr: &mut Vec<i32>) {
    let n = arr.len();

    for i in 0..n {
        let mut min_index = i;

        for j in i + 1..n {
            if arr[j] < arr[min_index] {
                min_index = j;
            }
        }

        if min_index != i {
            arr.swap(i, min_index);
        }
    }
}

fn main() {
    let mut numbers = vec![64, 25, 12, 22, 11];
    
    println!("Original array: {:?}", numbers);

    selection_sort(&mut numbers);

    println!("Sorted array: {:?}", numbers);
}
